llvm/include/llvm/ADT/FoldingSet.h

```diff
@@ -362,6 +362,21 @@ class FoldingSetNodeID {
     }
   }
 
+  void AddRanges(const SmallVector<std::pair<int64_t, int64_t>, 16> &Ranges) {
+    unsigned Size = Ranges.size();
+
+    unsigned NumsToInsert = 1 + 2 * Size;
+    Bits.reserve(Bits.size() + NumsToInsert);
+    Bits.push_back(Size);
+    if (Size == 0)
+      return;
+
+    for (const auto &[Start, End] : Ranges) {
```

Comment:
nit: const & doesn't do anything here
```suggestion
    for (auto [Start, End] : Ranges) {
```